### PR TITLE
Issue/PR Templates

### PR DESCRIPTION
**Pull request by DomT4**
Ref: https://github.com/blog/2111-issue-and-pull-request-templates

Let the debate begin! :smile:.

Edit - _Node are also having a [similar discussion](https://github.com/nodejs/node/pull/5291) that may be useful to keep an eye on given they also deal with a wad of Issue reports._

### CODE REVIEW

**Comment by tdsmith** [comment]
I think the shorter these are, the better (if only because my knee-jerk tendency to issue templates on other trackers is to find them annoying and a little condescending). I'm not sure casual Github users will understand checkbox Markdown.

For issue template, I might just write:
> If you're reporting a problem building a formula, please share your build logs by running `brew gist-logs name_of_formula` and paste the link it gives you here:

I don't know that PR templates are necessary unless you think there's a history of missing information in PR comments.

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -34,8 +34,4 @@ First time contributing to Homebrew? Read our [Code of Conduct](https://github.c
 * `git commit` with message formatted `foo: fix <insert details>`
 * [open a pull request](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/How-To-Open-a-Homebrew-Pull-Request-(and-get-it-merged).md#how-to-open-a-homebrew-pull-request-and-get-it-merged) and fix any failing tests
 
-### Report a security vulnerability
-
-* [Email security@brew.sh](mailto:security@brew.sh) (a private mailing list) with detailed reproduction instructions using [our PGP key](https://keybase.io/homebrew/key.asc)
-
```
Comment (tdsmith) [inline .github/CONTRIBUTING.md]: I think it's fine to leave this here.

Comment (DomT4) [inline .github/CONTRIBUTING.md]: We've literally stuck it on the front `README.md` so I'm not quite sure why we're duping it, but it seems more likely from our experience we'll see security Issues reported than security PRs filed.

**Comment by DomT4** [comment]
> I think the shorter these are, the better

I agree. I thought these were pretty brief as-is, but I'm happy to take culling suggestions :scissors:.

>  I'm not sure casual Github users will understand checkbox Markdown.

I stole the idea from Mike here, blame him :wink:.

> For issue template, I might just write:

It was discussed in chat the other day about cutting down on the amount of Feature/Formulae requests we leave sitting around indefinitely on the tracker. I thought it relatively important to squeeze it into there consequently, but again, happy to take comment on that discussion further.

> I don't know that PR templates are necessary unless you think there's a history of missing information in PR comments.

Most commonly:
* Submitting patches with no explanation/upstream submission.
* Submitting stuff that has obviously never been through a `brew audit`.
* Submitting stuff with commit messages that are so far off our style reference it's amusing.

The core stuff was a nod to things Baptiste and Mike have said on the core before.

**Comment by tdsmith** [comment]
To be clear, I'm suggesting that the issue template could be a single line of text.

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (mikemcquaid) [inline .github/CONTRIBUTING.md]: Agreed, I'd rather see it removed.

File: .gitignore [hunk 2]
```diff
@@ -38,7 +39,6 @@
 !/.travis.yml
 !/.yardopts
 !/CODEOFCONDUCT.md
-!/CONTRIBUTING.md
 !/LICENSE.txt
 !/README.md
 !/SUPPORTERS.md
```
Comment (mikemcquaid) [inline .gitignore]: Let's also consider moving this file out of the root and into the `share/doc/homebrew` while we're moving stuff around.

**Comment by mikemcquaid** [comment]
I agree that these could be a bit shorter but I think they are definitely better than nothing for now so let's :ship: and iterate.

**Comment by mikemcquaid** [comment]
Moved Kickstarter file in a9bf8bf too.

**Comment by DomT4** [comment]
Thanks Mike!

**Pull request closed without merging**